raft/src/main/java/org/apache/kafka/raft/internals/KRaftControlRecordStateMachine.java

```diff
@@ -65,7 +65,7 @@ public final class KRaftControlRecordStateMachine {
     //
     // 2. The read operations lastVoterSet, voterSetAtOffset and kraftVersionAtOffset read
     // the nextOffset first before reading voterSetHistory or kraftVersionHistory
-    private volatile long nextOffset = 0;
+    private volatile long nextOffset = -1;
```

Comment:
I think we should use the constant defined elsewhere in this PR for the initial log offset.

Reply:
Done. I redefined those values in ListenerContext and KRaftControlRecordStateMachine. I'll share the definition is a future PR when I refactor KafkaRaftClient and unified the controller/broker listener logic and the internal listener logic.

I didn't want to do this right away since the `git diff` would be too big. The controller/broker listeners are interested in committed state while the internal listener is interested in all of the state (committed and uncommitted).

Reply:
ok